New Native Gates page (#487)
Qiskit/documentation: docs/run/native-gates.mdx

| [switch_case](/api/qiskit/qiskit.circuit.SwitchCaseOp) | control flow for classical feedforward |
| [measure](/api/qiskit/qiskit.circuit.library.Measure) | |
| [delay](/api/qiskit/qiskit.circuit.Delay) |  |

### Falcon

| Name |  Notes |
| :--- | :--- |
| [CX](/api/qiskit/qiskit.circuit.library.CXGate)  | two-qubit gate |
| [RZ](/api/qiskit/qiskit.circuit.library.RZGate) | single-qubit gate |
| [ID](/api/qiskit/qiskit.circuit.library.IGate) | single-qubit gate wait cycle |
| [reset](/api/qiskit/qiskit.circuit.library.Reset)  | single-qubit gate, non-unitary; not the same as the initialization done at the start of a circuit to prepare the all 0's state |
| [if_else](/api/qiskit/qiskit.circuit.IfElseOp) | control flow for classical feedforward |
| [for_loop](/api/qiskit/qiskit.circuit.ForLoopOp) | control flow for classical feedforward |
| [switch_case](/api/qiskit/qiskit.circuit.SwitchCaseOp) | control flow for classical feedforward |
| [measure](/api/qiskit/qiskit.circuit.library.Measure) | |
| [delay](/api/qiskit/qiskit.circuit.Delay) |  |

<Admonition type="tip" title="The init_qubits flag">

> By default, the `init_qubits` flag is set to True. If the `init_qubits` flag is set to False, you must insert [resets](/api/qiskit/qiskit.circuit.library.Reset) at the start of your circuit to start your qubits in the ground state. Otherwise, if your job consists of multiple circuits, the qubits will begin in their state from the previous shot. (If a job consists of multiple circuits, the shots will be executed in a "round-robin" fashion. That is, one shot will be collected from every circuit, and then the next shot will be collected from every circuit, and so on.)

Reviewer: ```suggestion
The `init_qubits` flag controls whether qubits are reset to the zero state at the start of each circuit. Its default value is True, indicating that the qubits should be reset. If False, the qubits will begin in the final state from the previous shot, and you must manually insert [resets](/api/qiskit/qiskit.circuit.library.Reset) if you want to reset them to the zero state. If a job consists of multiple circuits, then the shots are executed in a "round-robin" fashion. That is, each circuit will be executed in sequence to obtain one shot from each circuit. This process is then repeated until the requested number of shots has been obtained from all circuits.
```

Reviewer: If we include this note, we have to mention what function the `init_qubits` flag is an argument of. A quick search of the source code suggests it is from [`assemble`](https://github.com/Qiskit/qiskit/blob/ec859e174ca2ac35f5c740895bf4040d5face47c/qiskit/compiler/assembler.py#L42). But is this function even mentioned elsewhere in the docs?

Author: Also, the Conditional Reset page (which has been removed) had this pph:

> **Note:** If `init_qubits` is set to false and you want qubits to start in the ground state, you must insert resets at the start of your circuit. Otherwise, qubits will begin with their state from the previous shot (after undergoing some thermalization process for the interval between shots as set by `rep_delay`). Note that the first shot of the first circuit may still be in the thermal ground state due to a longer delay existing during the loading of your job to hardware.

Author: So that's what I'm working with. Blake specifically mentioned the init_qubits flag, as long as it is supported by Sampler (which it is)... does it feel too much like it comes out of nowhere?

Reviewer 2: > From reading the note, it's not clear to me in what sense the `init_qubits` flag is supported by Sampler. Is it `Sampler(init_qubits=False)`?

You specify it as an execution option: ExecutionOptions(shots=4000, init_qubits=True https://docs.quantum.ibm.com/api/qiskit-ibm-runtime/qiskit_ibm_runtime.options.ExecutionOptions

Reviewer: Ah ok, the note should say that.

> The `init_qubits` flag controls whether qubits are reset to the zero state at the start of each circuit. Its default value is True, indicating that the qubits should be reset. If False, the qubits will begin in the final state from the previous shot, and you must manually insert [resets](/api/qiskit/qiskit.circuit.library.Reset) if you want to reset them to the zero state. If a job consists of multiple circuits, then the shots are executed in a "round-robin" fashion. That is, each circuit will be executed in sequence to obtain one shot from each circuit. This process is then repeated until the requested number of shots has been obtained from all circuits.

Reviewer 2: ```suggestion
The `init_qubits` flag, set as a [primitive execution option,](/api/qiskit-ibm-runtime/qiskit_ibm_runtime.options.ExecutionOptions) controls whether qubits are reset to the zero state at the start of each circuit. Its default value is True, indicating that the qubits should be reset. If False, the qubits will begin in the final state from the previous shot, and you must manually insert [resets](/api/qiskit/qiskit.circuit.library.Reset) if you want to reset them to the zero state. If a job consists of multiple circuits, then the shots are executed in a "round-robin" fashion. That is, each circuit will be executed in sequence to obtain one shot from each circuit. This process is then repeated until the requested number of shots has been obtained from all circuits.
```

Author: Thanks Becky!

</Admonition>

## Next steps

> <Admonition type="tip" title="Recommendations">

Reviewer 3: one of our goals is to have every docs page include a link to a learning platform topic where users can start using these concepts in practice, can you add a link to a sensible learning platform course/tutorial here please?

    - Read about basis gates in the [Represent quantum computers](../transpile/representing_quantum_computers#basis-gates) topic.

> </Admonition>

Author: ```suggestion
    - Try out the [Submit transpiled circuits](https://learning.quantum.ibm.com/tutorial/submit-transpiled-circuits) tutorial.
</Admonition>
```

| [measure](/api/qiskit/qiskit.circuit.library.Measure) | |
| [delay](/api/qiskit/qiskit.circuit.Delay) |  |

<Admonition type="tip" title="The init_qubits flag">

The `init_qubits` flag, set as a [primitive execution option,](/api/qiskit-ibm-runtime/qiskit_ibm_runtime.options.ExecutionOptions) controls whether qubits are reset to the zero state at the start of each circuit. Its default value is `True`, indicating that the qubits should be reset. If `False`, the qubits will begin in the final state from the previous shot, and you must manually insert [resets](/api/qiskit/qiskit.circuit.library.Reset) if you want to reset them to the zero state. If a job consists of multiple circuits, then the shots are executed in a "round-robin" fashion. That is, each circuit will be executed in sequence to obtain one shot from each circuit. This process is then repeated until the requested number of shots has been obtained from all circuits.
</Admonition>

## Next steps

<Admonition type="tip" title="Recommendations">
    - Read about basis gates in the [Represent quantum computers](../transpile/representing_quantum_computers#basis-gates) topic.

>     - Try out the [Submit transpiled circuits](https://learning.quantum.ibm.com/tutorial/submit-transpiled-circuits) tutorial.

Author: ```suggestion
    - Apply your knowledge of basis gates to one of these [workflow example tutorials.](https://learning.quantum.ibm.com/catalog/tutorials?category=workflow-example)"
```

>     - Apply your knowledge of basis gates to one of these [workflow example tutorials.](https://learning.quantum.ibm.com/catalog/tutorials?category=workflow-example)"

Author: ```suggestion
    - Apply your knowledge of basis gates to one of these [workflow example tutorials.](https://learning.quantum.ibm.com/catalog/tutorials?category=workflow-example)
```

---
title: Native gates and operations
description: Summary of the native gates and operations supported by IBM Quantum systems

---

# Native gates and operations

> Each [processor family](processor-types) has a native gate set, and the systems in each family support running only those particular gates and operations. Transpilation translates an arbitrary circuit to the native gate set supported by a specified system.  You can view the native gates and operations for a system either [with Qiskit](#native-gates-with-qiskit) or on the IBM Quantum Platform [Compute resources page](#native-gates-on-platform).

Reviewer 2: It feels weird to just have a link to Basis gates at the end when they've never been mentioned in the topic.  Maybe something like this: 

Each [processor family](processor-types) has a native gate set.  By default, the systems in each family only support running the gates and operations in the native gate set. Thus, every gate in the circuit must be translated (by the transpiler) to the elements of this set.   

You can view the native gates and operations for a system either [with Qiskit](#native-gates-with-qiskit) or on the IBM Quantum Platform [Compute resources page](#native-gates-on-platform).

Note: The terms _native gate set_ and _basis gates_ are often used interchangeably.  However, you can specify a different set of basis gates to use, while the native gate set never changes.  For information about changing the basis gates, see https://docs.quantum.ibm.com/transpile/representing_quantum_computers#basis-gates

Author: Very nice wording! Would you remove the Next steps section, or keep it as-is in addition to adding this note in the first section?
